contrib/babelfishpg_common/src/collation.c

```diff
@@ -1393,4 +1437,46 @@ get_collation_callbacks(void)
 		collation_callbacks_var.has_ilike_node = &has_ilike_node;
 	}
 	return &collation_callbacks_var;
+}
+
+/*
+ * babelfish_define_type_default_collation - would be used to update default collation of Babelfish data types
+ * to correct tsql collation.
+ */
+Oid
+babelfish_define_type_default_collation(Oid typeNamespace)
+{
+	/* We should only override the default collation for Babelfish data types. */
+	if (strcmp(get_namespace_name(typeNamespace), "sys") != 0)
+		return DEFAULT_COLLATION_OID;
+
+	/* 
+	 * If upgrade is going on then we should use oid corresponding to 
+	 * babelfishpg_tsql.restored_server_collation_name.
+	 */
+	if (babelfish_restored_server_collation_name != NULL)
+		return get_collation_oid_internal(babelfish_restored_server_collation_name);
+
+	get_server_collation_oid_internal(false); /* set and cache server_collation_oid */
+
+	if (OidIsValid(server_collation_oid))
+		return server_collation_oid;
+
+	return DEFAULT_COLLATION_OID;
+}
+
+PG_FUNCTION_INFO_V1(get_babel_server_collation_oid);
+
+/*
+ * get_babel_server_collation_oid - corresponding to sys.babelfishpg_tsql_get_babel_server_collation_oid() function
+ * which would be available and useful during upgrade only.
+ */
+Datum
+get_babel_server_collation_oid(PG_FUNCTION_ARGS)
+{
+	/* If babelfish_restored_server_collation_name is set then use it. */
+	if (babelfish_restored_server_collation_name != NULL)
```

Comment:
I thought approach was :

- set restore GUC using dump
- Use this GUC when creating types in MVU to have correct collation
- Later as part of minor version upgrade, skip collation correction if current value is not PG default

It seems the approach taken is:
- set GUC at database level during dump
- I think types will not get the correct collation as alter database needs restart
- Later during minor upgrade, use restore GUC to set correct value wherever needed

It seems first solution is simpler